BatteryConditions.cpp

```diff
@@ -0,0 +1,53 @@
+#include "BatteryConditions.h"
+#include "iostream"
+
+
+
+bool BatteryConditions::checkRange(float actualValue, float minValue, float maxValue)
+{
+    return (actualValue < minValue || actualValue > maxValue);
+}
+
+bool BatteryConditions::batteryIsOk(float temperature, float soc, float chargeRate)
+{
+    
+    return (TemperatureCheck(temperature) && SocCheck(soc)
+        && ChargeRateCheck(chargeRate));
+    
+}
+
+bool BatteryConditions::TemperatureCheck(int threshold)
+{
+    if (checkRange(threshold , 0 , 45)) {
+     std::cout << "Temperature out of range!\n";
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
+
+bool BatteryConditions::SocCheck(int threshold)
+{
+    if (checkRange(threshold, 20, 80) ){

```

Comment:
usage of hard coded values will lead to issues during maintenance. if it gets used in many places and then a change in one place needs to be consistent with all others. There is a scope for improvement here.